client/chat/action-creators.ts

```diff
@@ -15,7 +26,28 @@ export function joinChannel(channel: string): ThunkAction {
     })
     dispatch({
       type: '@chat/joinChannel',
-      payload: fetchJson<void>(apiUrl`chat/${channel}`, { method: 'POST' }),
+      payload: fetchJson<void>(apiUrl`chat/${channel}`, { method: 'POST' }).catch(err => {
+        // TODO(2Pac): Rework how joining channel works. Currently we first navigate to the channel
+        // and then attempt to join it. Which seems weird to me?
+        //
+        // To work around this for now, if the user is banned we redirect them to the index page,
+        // but ideally they wouldn't be navigated to the channel in the first place.
+        replace('/')
```

Comment:
I was thinking of doing this in a separate PR, since I also need to handle the case of someone landing in a channel by directly entering the URL of it.

Reply:
Yeah, that's fine